### PR TITLE
3 Window commands

### PR DESCRIPTION
**Pull request by Mark-Phillipson**
I have created 3 windows are custom commands.

Two are to enter curly brackets, with either leaving the cursor inside or outside.

The third command, is a custom keyboard shortcut I set up to focus the external terminal  (cmd.exe)

### CODE REVIEW

**Comment by awulkan** [comment]
I have some feedback.

1. Simply saying `focus terminal` already works on Windows 10. Your key combination does not work for me. There's also the command `focus cmd` if you're using the old command line. Or `focus powershell` if you're using that.
2. "curly brackets" are called "braces" in Serenade.
3. You should probably use `<%cursor%>` to specify where the cursor should end up at, instead of triggering the left or space key.

I personally don't think these commands are necessary. Serenade should probably be able to handle braces and brackets out of the box in all operating systems. The fact that it doesn't right now feels like a bug to me.

**Comment by Mark-Phillipson** [comment]
I cannot seem to get the <%cursor%> working with a global/api.typeText command.

I'm under the impression it will only work with the snippet command, but could be wrong.

In the UK braces is something used to keep your trousers up :-)

**Comment by awulkan** [comment]
> I'm under the impression it will only work with the snippet command, but could be wrong.

That might be true. I haven't tried using it in a command, so I just assumed it would work. Sorry.

File: system/windows.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+serenade.global().command("curly brackets in",async api =>{
+    await api.typeText("{}");
+    await api.pressKey("left")
+});
+serenade.global().command("curly brackets out",async api =>{
+    //await api.focus("terminal");
```
Comment (jjspace) [inline system/windows.js:6]: There's no need for this comment here which I believe is leftover from copying an example?

**Review by jjspace — COMMENTED** [review]
> I'm under the impression it will only work with the snippet command

That may be true but both of the braces "commands" really probably _should_ be snippets. Something like `curly braces in: "{<%cursor%>}"` and `curly braces out: "{} <%cursor%>"` (probably don't need cursor at all in the second).

I would also say that it's probably best to avoid adding _any_ commands that just perform a keyboard shortcut for now as there's absolutely no way to be sure it will perform the same way across systems. Things like this, especially since this one is (I believe) specific to vscode, should wait for actual API integration, or at the _very_ least not be a `.global()` level command. Direct access to VSCode methods has already been requested on [slack here](https://serenadecommunity.slack.com/archives/CG39QS27R/p1613700658099300)

Also see small code comment below

**Comment by Mark-Phillipson** [comment]
Cannot do a snippet without a language, so I guess these are nonstarters for global commands :-(

**Comment by tmacwill** [comment]
Thanks for opening this pull request!

That's a good point—in the current version of the API, you can't do snippets without a language, but we can make that possible, so these snippets could exist in any kind of file. I'll update this thread when that's available, so you can use snippet commands. By the way, you can also set up a custom pronunciation (https://serenade.ai/docs/#custom-pronunciations) if you're just looking to remap `brackets` to `braces`.

For the `terminal` command you'd created, is that to focus the terminal within VSCode, or does that focus the cmd.exe app? I hadn't seen that shortcut before!

**Comment by Mark-Phillipson** [comment]
Thanks.

With hindsight I realise the custom terminal command was using a custom keyboard shortcut to focus the internal terminal, that I had set up. Which is probably not much use to anybody else. It would be great if we could refer to actual commands using the Visual Studio code API for example. This would cut out the need for mapping to shortcuts altogether.

I will look into the custom pronunciations.

**Comment by awulkan** [comment]
It would be great if we could refer to actual commands using the Visual Studio code API for example. This would cut out the need for mapping to shortcuts altogether.

Yep, I agree with this, and it has been proposed earlier in the Slack chat.
I have custom commands to toggle the terminal as well, but they use Swedish keys, since that's the default mapping in VS Code for me. So I can't share them with the international community.

**Comment by tmacwill** [comment]
to follow up on this thread, you can now use `api.evaluateInPlugin` to run VS Code API commands directly, rather than needing to use `pressKey`! (https://serenade.ai/docs/api#reference)

**Pull request closed without merging**